src/back/spv/writer.rs

```diff
@@ -108,13 +120,28 @@ impl Writer {
         arena: &crate::Arena<crate::Type>,
         handle: crate::Handle<crate::Type>,
     ) -> Word {
-        match self.lookup_type.lookup_id(handle) {
-            Some(word) => word,
-            None => {
-                let (instruction, id) = self.parse_type_declaration(arena, handle);
-                instruction.to_words(&mut self.logical_layout.declarations);
-                id
-            }
+        let ty = &arena[handle];
+
+        if let Some(id) = self
+            .lookup_type_new
+            .iter()
+            .find(|&(_, v)| match v {
```

Comment:
once we reverse the table, the iteration here will go away, and a single lookup remain

Reply:
I agree with the idea of reversing the Lookups, but how will this iteration go away? Maybe I am not yet understanding what you mean, but as far as I can tell is that we still have to have the exact same code, but instead of matching against the value, we match against the key?

Reply:
The way I think this would work is that you construct the `LookupType::Handle(handle)` as a key, and do `lookup_types.entry(key)`. If it's `Occupied`, just return the value, which is SPIR-V ID. If it's vacant, do the `write_type_declaration` stuff, which produces a new ID, and fill the vacant entry with this.